polars/polars-io/src/csv/write_impl.rs

```diff
@@ -26,7 +36,7 @@ fn fmt_and_escape_str(f: &mut Vec<u8>, v: &str, options: &SerializeOptions) -> s
         }
         let surround_with_quotes = memchr2(options.delimiter, b'\n', v.as_bytes()).is_some();
```

Comment:
If we set this `options.string_field_enclosed || memchr2...` we don't have to do the expensive search if the `string_field_enclosed` value is set.